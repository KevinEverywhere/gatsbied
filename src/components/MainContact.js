import React from 'react'
import PropTypes from 'prop-types'

class MainContact extends React.Component {
  constructor(props){
    super();
  }
  render() {
    return (
      <main>
      Main Contact div
      </main>
    )
  }
}

MainContact.propTypes = {
  route: PropTypes.object,
  article: PropTypes.string,
  articleTimeout: PropTypes.bool,
  onCloseArticle: PropTypes.func,
  timeout: PropTypes.bool,
  setWrapperRef: PropTypes.func.isRequired,
}

export default MainContact
