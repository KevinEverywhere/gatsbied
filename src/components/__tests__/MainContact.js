import React from "react"
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from "react-test-renderer"
import MainContact from "../MainContact"

Enzyme.configure({ adapter: new Adapter() });

describe("MainContact", () => {
  it('MainContact: renders correctly', () => {
    const tree = renderer.create(<MainContact timeout={false} setWrapperRef={function(){}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('componentWillMount: should use initial state', () => {
    const wrapper = shallow(<MainContact timeout={false} setWrapperRef={function(){}}/>);
    expect(wrapper.instance().state.nextPage).toBe('ContactReply');
  });
  it('componentWillMount: should set the passed nextPage value to state', () => {
    const wrapper = shallow(<MainContact timeout={false} setWrapperRef={function(){}}/>);
    wrapper.instance().state.nextPage='ContactOops';
    expect(wrapper.instance().state.nextPage).toBe('ContactOops');
  });
  // it('incrementMainContact: should increment state.count by 1', () => {
  //   const wrapper = shallow(<MainContact initialCountValue={0}/>);
  //   const instance = wrapper.instance();
  //   expect(instance.state.count).toBe(0);
  //   instance.incrementMainContact();
  //   expect(instance.state.count).toBe(1);
  // });
  // it('decrementMainContact: should decrement state.count by 1', () => {
  //   const wrapper = shallow(<MainContact initialCountValue={1}/>);
  //   const instance = wrapper.instance();
  //   expect(instance.state.count).toBe(1);
  //   instance.decrementMainContact();
  //   expect(instance.state.count).toBe(0);
  // });
  // it('should call props on increment/decrement', () => {
  //   const incrementSpy = jest.fn();
  //   const decrementSpy = jest.fn();
  //   const wrapper = shallow(<MainContact initialCountValue={1} onIncrement={incrementSpy} onDecrement={decrementSpy}/>);
  //   const instance = wrapper.instance();
  //   instance.incrementMainContact();
  //   expect(incrementSpy).toBeCalledWith(2);
  //   instance.decrementMainContact();
  //   expect(decrementSpy).toBeCalledWith(1);
  // });
});
