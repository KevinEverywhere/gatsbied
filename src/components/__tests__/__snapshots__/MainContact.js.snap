// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContact MainContact: renders correctly 1`] = `
<main>
  <form
    method="post"
    name="contactForm"
    onSubmit={[Function]}
  >
    <div
      className="field half first"
    >
      <label
        htmlFor="uname"
      >
        Name 
        <span
          style={
            Object {
              "color": "white",
              "fontWeight": 900,
            }
          }
        />
      </label>
      <input
        id="uname"
        name="uname"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="field half"
    >
      <label
        htmlFor="email"
      >
        Email 
        <span
          style={
            Object {
              "color": "white",
              "fontWeight": 900,
            }
          }
        />
      </label>
      <input
        id="email"
        name="email"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="field"
    >
      <label
        htmlFor="message"
      >
        Message
      </label>
      <textarea
        id="message"
        name="message"
        onChange={[Function]}
        rows="6"
      />
    </div>
    <ul
      className="actions"
    >
      <li>
        <input
          className="special"
          name="submitter"
          type="submit"
          value="Send Message"
        />
      </li>
      <li>
        <input
          name="resetter"
          type="reset"
          value="Reset"
        />
      </li>
    </ul>
    <input
      name="_text"
      type="hidden"
      value="contactReply"
    />
  </form>
</main>
`;

exports[`MainContact renders correctly 1`] = `
<main>
  <form
    method="post"
    name="contactForm"
    onSubmit={[Function]}
  >
    <div
      className="field half first"
    >
      <label
        htmlFor="uname"
      >
        Name 
        <span
          style={
            Object {
              "color": "white",
              "fontWeight": 900,
            }
          }
        />
      </label>
      <input
        id="uname"
        name="uname"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="field half"
    >
      <label
        htmlFor="email"
      >
        Email 
        <span
          style={
            Object {
              "color": "white",
              "fontWeight": 900,
            }
          }
        />
      </label>
      <input
        id="email"
        name="email"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="field"
    >
      <label
        htmlFor="message"
      >
        Message
      </label>
      <textarea
        id="message"
        name="message"
        onChange={[Function]}
        rows="6"
      />
    </div>
    <ul
      className="actions"
    >
      <li>
        <input
          className="special"
          name="submitter"
          type="submit"
          value="Send Message"
        />
      </li>
      <li>
        <input
          name="resetter"
          type="reset"
          value="Reset"
        />
      </li>
    </ul>
    <input
      name="_text"
      type="hidden"
      value="contactReply"
    />
  </form>
</main>
`;
